File: app/src/main/java/com/esorokin/lantector/model/network/data/BaseResponse.java
package com.esorokin.lantector.model.network.data;

import org.simpleframework.xml.Element;

public class BaseResponse {
	@Element(name = "status")
	private String statusResult;

	@Element(name = "statusInfo", required = false)
	private String statusCode;

	protected BaseResponse() {
		//not available for directly create
	}

	public ApiStatusCode apiStatusCode() {
		return ApiStatusCode.from(statusCode);
	}

	public ApiStatusResult apiStatusResult() {
		return ApiStatusResult.from(statusResult);
	}
}

File: app/src/main/java/com/esorokin/lantector/model/network/data/BaseResponse.kt
package com.esorokin.lantector.model.network.data

import org.simpleframework.xml.Element

open class BaseResponse(
        @field:Element(name = "status")
        var statusResult: String = "ERROR",

        @field:Element(name = "statusInfo", required = false)
        var statusCode: String = "") {

    fun apiStatusResult(): ApiStatusResult = ApiStatusResult.from(statusResult)

    fun apiStatusCode(): ApiStatusCode = ApiStatusCode.from(statusCode)
}
